upgrades/upgrade.gd

```diff
@@ -0,0 +1,17 @@
+extends Resource
+class_name UpgradeDefinition
+
+@export var id: String
+@export var name: String
+@export var cost: int
+
+func _init(_id: String, _name: String, _cost: int) -> void:
+	self.id = _id
+	self.name = _name
+	self.cost = _cost
```

Comment:
Cost here is an int, but I would prefer it if these were functions instead, I didn't find good ways to type these, but I know we can store and pass functions around.

Still, in godot it might be more idiomatic to subclass it, I'll have to think that through.